tests/recognizer-tests.js

```diff
@@ -129,6 +129,16 @@ test("A dynamic route recognizes", function() {
   equal(router.recognize("/zoo/baz"), null);
 });
 
+test("A dynamic route with encoded params", function() {
+  var handler = {};
+  var router = new RouteRecognizer();
+  router.add([{ path: "/foo/:bar", handler: handler }]);
+
+  equal(router.recognize("/foo/bar%20baz")[0].params.bar, "bar baz");
+  equal(router.recognize("/foo/bar%23baz")[0].params.bar, "bar#baz");
+  equal(router.recognize("/foo/bar%3fbaz")[0].params.bar, "bar?baz");
+});
+
```

Comment:
I can't make this *and* the following test both pass:
```js
test("A route with URL-encoded parts recognizes", function() {
  var handler = {};
  var router = new RouteRecognizer();
  router.add([{ path: "/foo%20bar%3fbaz", handler: handler }]);

  resultsMatch(router.recognize("/foo%20bar%3fbaz"), [{ handler: handler, params: {}, isDynamic: false }]);
});
```
though the following variant does:
```js
test("A route with URL-encoded parts recognizes", function() {
  var handler = {};
  var router = new RouteRecognizer();
  router.add([{ path: "/foo bar?baz", handler: handler }]);

  resultsMatch(router.recognize("/foo%20bar%3fbaz"), [{ handler: handler, params: {}, isDynamic: false }]);
});
```
This PR therefore would mean that the `path` argument should be decoded.